File: 백준/Bronze/2884. 알람 시계/알람 시계.cc
#include <stdio.h>

int main()
{
    int hour, minute;
    scanf("%d %d", &hour, &minute);
    
    if(hour >= 0 && minute >= 45)
        printf("%d %d", hour, minute - 45);
    else if(hour == 0 && minute < 45)
        printf("23 %d", 60 + (minute - 45));
    else if(hour > 0 && minute < 45)
        printf("%d %d", hour - 1, 60 + (minute - 45));
    return 0;
        
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);
        int hour = scanner.nextInt();
        int minute = scanner.nextInt();
        
        if (minute >= 45) {
            System.out.println(hour + " " + (minute - 45));
        } else if (hour == 0) {
            System.out.println("23 " + (minute + 15));
        } else {
            System.out.println((hour - 1) + " " + (minute + 15));
        }

        scanner.close();
    }
}